File: trip-kit/src/main/java/com/skedgo/android/tripkit/OutOfRegionsException.java
package com.skedgo.android.tripkit;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.skedgo.android.common.model.Location;

public final class OutOfRegionsException extends RuntimeException {
  private final Location location;

  public OutOfRegionsException(@NonNull Location location, @Nullable String detailMessage) {
    super(detailMessage);
    this.location = location;
  }

  @NonNull
  public Location getLocation() {
    return location;
  }
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/OutOfRegionsException.kt
package com.skedgo.tripkit;

import androidx.annotation.Nullable;

public final class OutOfRegionsException extends RuntimeException {
    private final double latitude;
    private final double longitude;

    public OutOfRegionsException(
        @Nullable String detailMessage,
        double latitude,
        double longitude) {
        super(detailMessage);
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double latitude() {
        return latitude;
    }

    public double longitude() {
        return longitude;
    }
}
